Description: Fix ruby tests for older Ruby 1.8.x
 ruby/smoketest.rb: Older Ruby 1.8.x doesn't have Dir.mktmpdir() -
 just skip the compaction tests for these versions.
Author: Olly Betts <olly@survex.com>
Origin: upstream
Last-Update: 2011-05-17

--- xapian-bindings-1.2.5.orig/ruby/smoketest.rb
+++ xapian-bindings-1.2.5/ruby/smoketest.rb
@@ -242,6 +242,10 @@ class XapianSmoketest < Test::Unit::Test
   end
 
   def test_016_compactor
+    if ! Dir.respond_to?("mktmpdir")
+      # Older Ruby 1.8.x doesn't have Dir.mktmpdir() - just skip if so.
+      return
+    end
     Dir.mktmpdir("smokerb") {|tmpdir|
         db1path = "#{tmpdir}db1"
         db2path = "#{tmpdir}db2"
